src/plugins/badge.coffee

```diff
@@ -0,0 +1,648 @@
+  }
+
+  @-ms-keyframes fadein {
+    from { opacity: 0; }
+    to   { opacity: 1; }
+  }
+
+  @-o-keyframes fadein {
+    from { opacity: 0; }
+    to   { opacity: 1; }
+  }
+
+  .spinner::before {
+    content: "";
+    width: 48px;
+    height: 48px;
+    border-radius: 50%;
+    border: 4px solid rgba(255, 255, 255, .5);
+    border-top-color: rgba(246, 139, 36, .7);
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    margin-top: -24px;
+    margin-left: -24px;
+    animation: spin 1s linear infinite;
+    z-index: 1;
+  }
+
+  #sa-badge-floating-plugin {
+    all: initial;
+
+    display: block;
+
+    position: fixed;
+    bottom: 25px;
+    #{switch configuration.position
+      when 'bottom-left' then 'left: 25px;'
+      when 'bottom-right' then 'right: 25px;'}
+
+    width: 90px;
+    height: 90px;
+
+    -moz-border-radius: 49px;
+    -webkit-border-radius: 49px;
+    border-radius: 49px;
+
+    background-image: url('#{asset_url("badge/floating/large/theme/#{configuration.theme}/logo_@@flavor.png")}');
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: 90px 90px;
+
+    cursor: pointer;
+
+    z-index: 2147483647;
+
+    -webkit-animation: fadein 0.3s ease-in;
+    -moz-animation: fadein 0.3s ease-in;
+    -ms-animation: fadein 0.3s ease-in;
+    -o-animation: fadein 0.3s ease-in;
+    animation: fadein 0.3s ease-in;
+
+    box-shadow: 0 0 4px rgba(0,0,0,.14), 0 4px 8px rgba(0,0,0,.28);
+  }
+
+  #sa-badge-floating-plugin * {
+    all: unset;
+  }
+
+  #sa-badge-floating-plugin #sa-badge-floating-stars-container {
+    position: absolute;
+    overflow: visible;
+
+    bottom: 18px;
+    left: 20px;
+  }
+
+  #sa-badge-floating-plugin #sa-badge-floating-stars-container .sa-badge-star {
+    display: inline-block;
+    margin-right: -3px;
+
+    width: 9px;
+    height: 8px;
+
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: 9px 8px;
+  }
+
+  #sa-badge-floating-plugin #sa-badge-floating-stars-container .sa-badge-full-star {
+    background-image: url('#{asset_url("badge/floating/large/theme/#{configuration.theme}/star_full.png")}');
+  }
+
+  #sa-badge-floating-plugin #sa-badge-floating-stars-container .sa-badge-half-star {
+    background-image: url('#{asset_url("badge/floating/large/theme/#{configuration.theme}/star_half.png")}');
+  }
+
+  #sa-badge-floating-plugin #sa-badge-floating-stars-container .sa-badge-empty-star {
+    background-image: url('#{asset_url("badge/floating/large/theme/#{configuration.theme}/star_empty.png")}');
+  }
+
+  #sa-badge-modal {
+    position: fixed;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 2147483647;
+    visibility: visible;
+    overflow-y: auto;
+    padding: 50px 0;
+    opacity: 1;
+    background: rgba(0, 0, 0, .5);
+    -moz-user-select: -moz-none; // Disable text highlight in Firefox
+    -moz-user-select: none;      // Disable text highlight in Firefox
```

Comment:
We are not going to have a text inside the modal, except i am missing something.